src/components/side-panel/layer-panel/add-layer-button.js

```diff
@@ -0,0 +1,133 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React, {useState} from 'react';
+import styled from 'styled-components';
+import {FormattedMessage} from 'localization';
+
+import Tippy from '@tippyjs/react';
+import DropdownList from 'components/common/item-selector/dropdown-list';
+import {Add} from 'components/common/icons';
+import {Button} from 'components/common/styled-components';
+
+const DropdownContainer = styled.div.attrs({
+  className: 'add-layer-menu-dropdown'
+})`
+  .list-selector {
+    border-top: 0;
+    width: max-content;
+    padding: 8px 0;
+    /* disable scrolling, currently set to 280px internally */
+    max-height: unset;
+  }
+
+  .list__header {
+    padding: 8px 20px;
+  }
+
+  .list__item {
+    padding: 8px 20px;
+
+    &:hover {
+      background: ${props => `${props.theme.PURPLE2}22`};
+    }
+    &:hover > div {
+      color: ${props => props.theme.PURPLE2};
+    }
+  }
+
+  .list__item > div {
+    display: flex;
+    align-items: center;
+    flex-direction: row;
+    justify-content: flex-start;
+    line-height: 18px;
+    padding: 0;
+
+    svg {
+      margin-right: 10px;
+    }
+  }
+`;
+
+const AddLayerMenu = styled.div.attrs({
+  className: 'add-layer-menu'
+})`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  position: absolute;
+  top: 100%;
+  left: -53px;
+  z-index: 5;
+`;
+
+function AddLayerButtonFactory() {
+  const AddLayerButton = props => {
+    const {datasets, onOptionSelected} = props;
+    const [showAddLayerDropdown, setShowAddLayerDropdown] = useState(false);
+
+    const toggleAddLayerDropdown = () => {
+      setShowAddLayerDropdown(!showAddLayerDropdown);
+    };
+
+    return (
+      <Tippy
+        visible={showAddLayerDropdown}
+        arrow={false}
+        interactive
+        placement="bottom"
+        appendTo="parent"
+        duration={0}
+        content={
+          <div>
+            <AddLayerMenu>
+              <DropdownContainer>
+                <DropdownList
+                  displayOption={d => d.label}
+                  options={Object.values(datasets)}
+                  onOptionSelected={dataset => {
+                    onOptionSelected(dataset);
+                    setShowAddLayerDropdown(false);
+                  }}
+                />
+              </DropdownContainer>
+            </AddLayerMenu>
+          </div>
+        }
+      >
+        <Button
+          tabIndex={-1}
+          onBlur={() => setShowAddLayerDropdown(false)}
```

Comment:
create a callback for this as well

Reply:
Similar to above but maybe this is just too simple for a separate function.

Reply:
same